src/services/db/RepoService.ts

```diff
@@ -0,0 +1,247 @@
+import { AxiosCacheInstance } from "axios-cache-interceptor"
+
+import config from "@config/config"
+
+import logger from "@logger/logger"
+
+import UserWithSiteSessionData from "@root/classes/UserWithSiteSessionData"
+import { GitDirectoryItem, GitFile } from "@root/types/gitfilesystem"
+
+import GitFileSystemService from "./GitFileSystemService"
+import { GitHubService } from "./GitHubService"
+import * as ReviewApi from "./review"
+
+const WHITELISTED_GIT_SERVICE_REPOS = config.get(
+  "featureFlags.ggsWhitelistedRepos"
+)
+
+export default class RepoService extends GitHubService {
+  private readonly gitFileSystemService: GitFileSystemService
+
+  constructor(
+    axiosInstance: AxiosCacheInstance,
+    gitFileSystemService: GitFileSystemService
+  ) {
+    super({ axiosInstance })
+    this.gitFileSystemService = gitFileSystemService
+  }
+
+  isRepoWhitelisted(repoName: string): boolean {
+    return WHITELISTED_GIT_SERVICE_REPOS.split(",").includes(repoName)
+  }
+
+  getCommitDiff(siteName: string, base?: string, head?: string) {
+    return ReviewApi.getCommitDiff(siteName, base, head)
+  }
+
+  createPullRequest(
+    siteName: string,
+    title: string,
+    description?: string,
+    base?: string,
+    head?: string
+  ) {
+    return ReviewApi.createPullRequest(siteName, title, description, base, head)
+  }
+
+  getPullRequest(siteName: string, pullRequestNumber: number) {
+    return ReviewApi.getPullRequest(siteName, pullRequestNumber)
+  }
+
+  getBlob(repo: string, path: string, ref: string) {
+    return ReviewApi.getBlob(repo, path, ref)
+  }
+
+  updatePullRequest(
+    siteName: string,
+    pullRequestNumber: number,
+    title: string,
+    description?: string
+  ) {
+    return ReviewApi.updatePullRequest(
+      siteName,
+      pullRequestNumber,
+      title,
+      description
+    )
+  }
+
+  closeReviewRequest(siteName: string, pullRequestNumber: number) {
+    return ReviewApi.closeReviewRequest(siteName, pullRequestNumber)
+  }
+
+  mergePullRequest(siteName: string, pullRequestNumber: number) {
+    return ReviewApi.mergePullRequest(siteName, pullRequestNumber)
+  }
+
+  approvePullRequest(siteName: string, pullRequestNumber: number) {
+    return ReviewApi.approvePullRequest(siteName, pullRequestNumber)
+  }
+
+  async getComments(siteName: string, pullRequestNumber: number) {
```

Comment:
why do we need to re define the methods from super class? When extending, these methods will be available on RepoService right?

Is it for future modifications?

Reply:
Yup, this is for future modifications. Eventually, we should move towards having all of these functions having type definitions, but for now it's just an explicit passthrough to GitHubService.